cpp/src/utilities/grid_1d.cu

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/utilities/grid_1d.cuh>
+#include <cudf/detail/utilities/integer_utils.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+
+namespace cudf::detail {
+
+grid_1d::grid_1d(thread_index_type overall_num_elements,
+                 thread_index_type num_threads_per_block,
+                 thread_index_type elements_per_thread)
+  : num_threads_per_block(num_threads_per_block),
+    num_blocks(
+      util::div_rounding_up_safe(overall_num_elements, elements_per_thread * num_threads_per_block))
+{
+  CUDF_EXPECTS(num_threads_per_block > 0, "num_threads_per_block must be > 0");
+  CUDF_EXPECTS(num_blocks > 0, "num_blocks must be > 0");
+}
```

Comment:
This is never called from device code so I think it should be safe to make this a .cpp file.
If there is an issue including the `grid_1d.cuh` here then it may be possible to add something like `include <cuda_runtime.h>` to the .cuh file.

Reply:
Right, but this file does not call them. It should be possible to compile with the g++ compiler. The `<cuda_runtime.h>` header usually helps the host compiler ignore the device functions.

Reply:
Ok, thanks for trying.